src/Core/Framework/Api/EventListener/JsonRequestTransformerListener.php

```diff
@@ -23,7 +23,8 @@ public static function getSubscribedEvents(): array
 
     public function onRequest(RequestEvent $event): void
     {
-        if ($event->getRequest()->getContent() && stripos($event->getRequest()->headers->get('Content-Type', ''), 'application/json') === 0) {
+        // It's important to check the content-type before, otherwise we read the content
+        if (stripos($event->getRequest()->headers->get('Content-Type', ''), 'application/json') === 0 && $event->getRequest()->getContent()) {
```

Comment:
because we check first getContent, we load the full uploaded file into memory.

Reply:
you could use `str_starts_with` there if you wanted